### ISSUE TITLE
Nested Playlist Support

### ISSUE THREAD

**Issue by EnderShadow** [quodlibet/quodlibet#3910]
The proposed change
-------------------
Support for nested playlists. This would allow you to add an existing playlist to another playlist and instead of adding the songs from that playlist to the new playlist, it would reference the old playlist. For example, if I have playlists A and B, I might also want a playlist C which is playlist A and B where modifying A or B automatically updates C.

This is different from the search syntax that currently exists since playlist C should be ordered so that playlist A's contents are first in the same order as in A followed by B's contents in the same order as in B.

Benefits of this change
-----------------------
The benefits of this change are that it allows for an easy way to manage a hierarchy of playlists in an organized fashion. Aside from benefiting me since I have this use case, this would also benefit numerous others who have looked for music players with the same feature set.

Any issues this might cause
---------------------------
This should have a minimal impact on performance and shouldn't cause any issues except potentially when modifying a nested playlist which you're listening to.

There are a few ways to implement this, but this will either require a second type of playlist to be supported or migration to a new playlist format which can support both songs and playlists.

One potential hazard if this is implemented is self-referential playlists, so a check for that when loading playlists will likely need to be performed.

**Comment by EnderShadow** [comment]
A possible alternative to this might be to augment the search functionality with some sort of ordering system so that this can be achieved with the existing search system.

**Comment by bassstorm** [comment]
Such a change must be a massive re-do of code base... At the same time, I'm afraid there would be no any noticeable number of demanding users.
Anyway, let's listen for people here.

**Comment by jej** [comment]
I almost never use playlists with media players. In the past it was to manage batch of tracks to upload on a Rockbox player or usb keys. But since few years I don't need this anymore. So I am very far from the need for nested playlists...
Maybe this feature could be partially implemented with a plugin, but why and who is still the question.

**Comment by EnderShadow** [comment]
> Maybe this feature could be partially implemented with a plugin, but why and who is still the question.

As I mentioned above I generally like listening to my songs in a particular order and I use playlists to set up that order. I also usually organize it with multiple layers of nesting. This feature resolves the pain point and of adding new songs to multiple different playlists, especially when adding it to the middle of a playlist that has hundreds or thousands of songs.

**Comment by Meriipu** [comment]
Might maybe be easier to make an external tool to generate/regenerate the playlists for quodlibet and have that external tool have abstractions such as nesting in its configs.

**Comment by declension** [comment]
Yes, this is a big rethink of a few things, and not just the playlist browser as the songlist is shared between all these browsers so I'm not currently feeling the pay-off here.

In the meantime, there's #1065 that whilst not solving this does allow more organisation of playlists themselves
